### PR TITLE
authz: Clear stale Perforce external accounts when email unverified

### PR DESCRIPTION
**Pull request by ryanslade**
Closes https://github.com/sourcegraph/sourcegraph/issues/44482

Test plan:

Tests added

### CODE REVIEW

File: cmd/frontend/backend/user_emails.go [hunk 1]
```diff
@@ -180,7 +180,7 @@ func (e *userEmails) SetPrimaryEmail(ctx context.Context, userID int32, email st
 }
 
 // SetVerified sets the supplied e-mail as the verified email for the given user.
```
Comment (sashaostrikov) [inline cmd/frontend/backend/user_emails.go:182]: Let's update the doc as well, mentioning that an email is deleted when `!verified`

File: cmd/frontend/backend/user_emails_test.go [hunk 2]
```diff
@@ -450,9 +451,10 @@ func TestDeleteStaleExternalAccount(t *testing.T) {
 
 	svc := NewUserEmailsService(db, logger)
 	ctx = actor.WithInternalActor(ctx)
-	assert.NoError(t, svc.Add(ctx, createdUser.ID, email2))
 
-	t.Run("Perforce", func(t *testing.T) {
+	setup := func() {
+		require.NoError(t, svc.Add(ctx, createdUser.ID, email2))
```
Comment (sashaostrikov) [inline cmd/frontend/backend/user_emails_test.go:456]: Do we need to call `t.Helper()` in this function?

**Review by sashaostrikov — APPROVED** [review]
LGTM!

File: cmd/frontend/backend/user_emails_test.go (hunk 2 above)
Comment (ryanslade) [reply cmd/frontend/backend/user_emails_test.go:456]: If there's an error somewhere in that function we want to see the line number where it happens. If we `t.Helper()` the line number is skipped.

**Pull request merged**